MPSGameLift/Code/Source/MatchmakingSystemComponent.cpp

```diff
@@ -0,0 +1,235 @@
+                    return reader.Accept(skill);
+                }
+                return reader.Ignore();
+            }
+
+            int skill;
+        };
+
+        struct PlayerAttributes
+        {
+            bool OnJsonKey(const char* key, AWSCore::JsonReader& reader)
+            {
+                if (strcmp(key, "skill") == 0)
+                {
+                    return reader.Accept(skill);
+                }
+                return reader.Ignore();
+            }
+
+            PlayerSkill skill;
+        };
+
+        struct Latencies
+        {
+            bool OnJsonKey(const char* key, AWSCore::JsonReader& reader)
+            {
+                return reader.Accept(latencies[key]);
+            }
+
+            AZStd::unordered_map<AZStd::string, int> latencies;
+        };
+
+        struct Player
+        {
+            bool OnJsonKey(const char* key, AWSCore::JsonReader& reader)
+            {
+                if (strcmp(key, "LatencyInMs") == 0)
+                {
+                    return reader.Accept(latencies);
+                }
+                if (strcmp(key, "PlayerId") == 0)
+                {
+                    return reader.Accept(playerId);
+                }
+                if (strcmp(key, "Team") == 0)
+                {
+                    return reader.Accept(team);
+                }
+                if (strcmp(key, "PlayerAttributes") == 0)
+                {
+                    return reader.Accept(playerAttributes);
+                }
+                return reader.Ignore();
+            }
+
+            AZStd::string playerId;
+            AZStd::string team;
+            Latencies latencies;
+            PlayerAttributes playerAttributes;
+        };
+
+        //! Struct for storing the success response.
+        struct RequestMatchmakingResponse
+        {
+            bool OnJsonKey(const char* key, AWSCore::JsonReader& reader)
+            {
+                if (strcmp(key, "TicketId") == 0)
+                {
+                    return reader.Accept(ticketId);
+                }
+                if (strcmp(key, "Players") == 0)
+                {
+                    return reader.Accept(players);
+                }
+
+                return reader.Ignore();
+            }
+
+            AZStd::string ticketId;
+            AZStd::vector<Player> players;
+        };
+
+        //! Struct for storing the error.
+        struct RequestMatchmakingError
+        {
+            bool OnJsonKey(const char* key, AWSCore::JsonReader& reader)
+            {
+                if (strcmp(key, "message") == 0)
+                {
+                    return reader.Accept(message);
+                }
+
+                if (strcmp(key, "type") == 0)
+                {
+                    return reader.Accept(type);
+                }
+
+                return reader.Ignore();
+            }
+
+            //! Do not rename the following members since they are expected by the AWSCore dependency.
```

Comment:
Does AWSCore already have a struct that encapsulates these members? Maybe instead of making a specific RequestMatchmakingError struct, you can reuse that struct?

Reply:
Fixed! Thank you... also updated the matchmaking AWS Lambda to send back an error that works with the built in AWSCore::Error